### PR TITLE
Make MethodHandles in Jazzer final

### PR DESCRIPTION
**Pull request by fmeum**
This unlocks JIT optimizations.

### CODE REVIEW

File: agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java [hunk 1]
```diff
@@ -23,68 +23,83 @@
...
+    MethodHandle autofuzzFunction1Local = null;
+    MethodHandle autofuzzFunction2Local = null;
+    MethodHandle autofuzzFunction3Local = null;
+    MethodHandle autofuzzFunction4Local = null;
+    MethodHandle autofuzzFunction5Local = null;
+    MethodHandle autofuzzConsumer1Local = null;
+    MethodHandle autofuzzConsumer2Local = null;
+    MethodHandle autofuzzConsumer3Local = null;
+    MethodHandle autofuzzConsumer4Local = null;
+    MethodHandle autofuzzConsumer5Local = null;
```
Comment (kyakdan) [inline agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java:59]: Could you clarify why we need the temporary variables? IISC, they are assigned once and we could therefore use the final fields directly

Comment (fmeum) [reply agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java:59]: We unfortunately can't due to the checked exceptions: All finals have to end up initialized in all code paths, including if an exception is thrown in the middle. The only alternative would be to extract the individual initializations into functions.

File: agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java [hunk 2]
```diff
@@ -23,68 +23,83 @@
  * Helper class with static methods that interact with Jazzer at runtime.
  */
 final public class Jazzer {
-  private static Class<?> jazzerInternal = null;
+  private static final Class<?> jazzerInternal;
```
Comment (kyakdan) [inline agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java:26]: According to Google's style guide, static final fields should use upper shake case https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names.

File: agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java (hunk 1 above)
Comment (kyakdan) [reply agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java:59]: I see. Thanks!

File: agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java (hunk 2 above)
Comment (fmeum) [reply agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java]: Thanks, this also neatly solves the naming problem.

**Review by kyakdan — APPROVED** [review]

**Pull request merged**